src/main/java/org/assertj/guava/api/RangeSetAssert.java

```diff
@@ -0,0 +1,982 @@
+   * </p>
+   *
+   * @param rangeSet the range set with ranges to check whether the actual {@code RangeSet} intersects at least one of
+   *                 them.
+   * @return this {@link RangeSetAssert} for assertions chaining.
+   * @throws AssertionError if the actual {@code RangeSet} is {@code null}.
+   * @throws AssertionError if the actual {@code RangeSet} does not intersect any of the ranges from the given ranges
+   *         set.
+   * @throws IllegalArgumentException if range set is null or it is empty while actual is not empty.
+   */
+  public RangeSetAssert<T> intersectsAnyElementsOf(final RangeSet<T> rangeSet) {
+    rangeSets.assertIntersectsAnyElementsOf(info, actual, rangeSet);
+    return myself;
+  }
+
+  /**
+   * Verifies that the given {@code RangeSet} does not intersect the given ranges.
+   * <p>
+   * Example:
+   *
+   * <pre><code class='java'>
+   * RangeSet&lt;Integer&gt; rangeSet = TreeRangeSet.create();
+   *
+   * rangeSet.add(Range.closed(0, 100));
+   * rangeSet.add(Range.closed(200, 300));
+   * rangeSet.add(Range.closed(500, 600));
+   *
+   * assertThat(rangeSet).doesNotIntersect(Range.closed(120, 150),
+   *                                       Range.open(302, 490),
+   *                                       Range.open(600, 670));</code></pre>
+   * </p>
+   *
+   * @param ranges the ranges to check whether the actual {@code RangeSet} does not intersect them.
+   * @return this {@link RangeSetAssert} for assertions chaining.
+   * @throws AssertionError if the actual {@code RangeSet} is {@code null}.
+   * @throws AssertionError if the actual {@code RangeSet} intersects the given ranges.
+   * @throws IllegalArgumentException if ranges are null or ranges are empty while actual is not empty.
+   */
+  public RangeSetAssert<T> doesNotIntersect(final Range<T>... ranges) {
+    rangeSets.assertDoesNotIntersect(info, actual, ranges);
+    return myself;
+  }
+
+  /**
+   * Verifies that the given {@code RangeSet} does not intersect ranges from the given range set.
+   * <p>
+   * Example:
+   *
+   * <pre><code class='java'>
+   * RangeSet&lt;Integer&gt; rangeSet = TreeRangeSet.create();
+   *
+   * rangeSet.add(Range.closed(0, 100));
+   * rangeSet.add(Range.closed(200, 300));
+   * rangeSet.add(Range.closed(500, 600));
+   *
+   * assertThat(rangeSet).doesNotIntersect(ImmutableRangeSet.of(Range.close(120, 170)));</code></pre>
+   * </p>
+   *
+   * @param rangeSet the range set to check whether the actual {@code RangeSet} does not intersect ranges from it.
+   * @return this {@link RangeSetAssert} for assertions chaining.
+   * @throws AssertionError if the actual {@code RangeSet} is {@code null}.
+   * @throws AssertionError if the actual {@code RangeSet} intersects the ranges from the given range set.
+   * @throws IllegalArgumentException if range set is null or it is empty while actual is not empty.
+   */
+  public RangeSetAssert<T> doesNotIntersectAll(final RangeSet<T> rangeSet) {
```

Comment:
rename to `doesNotIntersectAnyRangeFrom`

Reply:
done